Objects/object.c

```diff
@@ -239,9 +242,8 @@ PyObject_InitVar(PyVarObject *op, PyTypeObject *tp, Py_ssize_t size)
     if (op == NULL)
         return (PyVarObject *) PyErr_NoMemory();
     /* Any changes should be reflected in PyObject_INIT_VAR */
-    op->ob_size = size;
-    Py_TYPE(op) = tp;
-    _Py_NewReference((PyObject *)op);
+    Py_SIZE(op) = size;
+    PyObject_Init((PyObject *)op, tp);
```

Comment:
Shouldn't this just call `_PyObject_INIT()` ?

Reply:
Or, actually, shouldn't `PyObject_Init()` just call `_PyObject_INIT()` ? Duplicating that code across two files seems unnecessary, maybe even a bit error prone.

Reply:
`PyObject_Init` is defined in the same file, so the compiler should have everything it needs to inline as it sees fit.

As for code duplication, that's a valid point, but I'd call that out of scope for this PR. Let's keep orthogonal issues revertable separately.